game/gameboard.py

```diff
@@ -239,23 +265,25 @@ def _winner_in_array(self, testing_array):
 
     def _check_rows_for_winner(self):
         """Check if there's a winner row-wise"""
-        for i in range(self.ROWSCOUNT):
-            if self._winner_in_array(self._matrix[i]):
+        rows_iter = self.get_rows()
+        for row in rows_iter:
+            if self._winner_in_array(numpy.asarray(row)):
                 return True
         return False
 
     def _check_columns_for_winner(self):
         """Check if there's a winner column-wise"""
-        for i in range(self.COLUMNSCOUNT):
-            if self._winner_in_array(self._retrieve_column(i)):
+        columns_iter = self._get_columns()
+        for column in columns_iter:
+            if self._winner_in_array(numpy.asarray(column)):
                 return True
         return False
 
     def _check_diagonals_for_winner(self):
         """Check if there's a winner diagonal-wise"""
-        for i in range(-2, 4):
-            if self._winner_in_array(self._matrix.diagonal(i)) or \
-               self._winner_in_array(numpy.flipud(self._matrix).diagonal(i)):
+        diag_iter = self._get_diagonals()
+        for diagonal in diag_iter:
```

Comment:
I really don't see the need for the `*_iter` variables i the methods above... Why not just `for diagonal in self._get_diagonals():`?